python/ml_croissant/_src/operation_graph/operations/extract.py

```diff
@@ -0,0 +1,53 @@
+"""Extract operation module."""
+
+import dataclasses
+import fnmatch
+import logging
+import os
+import re
+import tarfile
+
+from etils import epath
+from ml_croissant._src.operation_graph.base_operation import Operation
+from ml_croissant._src.operation_graph.operations.download import (
+    get_download_filepath,
+    get_hash,
+)
+from ml_croissant._src.nodes import (
+    FileObject,
+    FileSet,
+)
+from ml_croissant._src.paths import EXTRACT_PATH
+import pandas as pd
+
+
+@dataclasses.dataclass(frozen=True, repr=False)
+class Untar(Operation):
+    """Un-tars "application/x-tar" and yields filtered lines."""
+
+    node: FileObject
+    target_node: FileObject | FileSet
+
+    def __call__(self):
+        includes = fnmatch.translate(self.target_node.includes)
+        url = self.node.content_url
+        archive_file = get_download_filepath(url)
+        hashed_url = get_hash(url)
+        extract_dir = EXTRACT_PATH / hashed_url
+        if not extract_dir.exists():
+            with tarfile.open(archive_file) as tar:
+                tar.extractall(extract_dir)
+        logging.info(
+            "Found directory where data is extracted: %s.", os.fspath(extract_dir)
```

Comment:
Nit: maybe "The data has been extracted to: %s" ("Found directory... " sounds a bit vague to me)